File: Arrays101/1295. Find Numbers with Even Number of Digits.java
class Solution {
    public int findNumbers(int[] nums) {
        int count = 0;
        for(int i=0;i<nums.length;i++){
            if((int)(Math.log10(nums[i]))%2 == 1) count++;
            //remember that you're counting evens actually
        }
        return count;
    }
}
/*
huh, this is surprisingly slower than the math solution
class Solution {
    public int findNumbers(int[] nums) {
        int count = 0;
        for(int i=0;i<nums.length;i++){
            if(String.valueOf(nums[i]).length() %2 == 0) count++;
        }
        return count;
    }
}
*/

File: Arrays101/1295. Find Numbers with Even Number of Digits.py
import math

class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        count = 0
        for x in nums:
            if(int(math.log10(x))%2): count += 1 # you're counting every odd, so you want to move the number by 1
            
        return count
        
        """
class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        return len([x for x in nums if len(str(x)) % 2 == 0])

        list comprehension is approximately as fast in this usecase
        """        
        